web/src/Web.App/Attributes/SchoolAuthorizationAttribute.cs

```diff
@@ -0,0 +1,20 @@
+﻿using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Web.App.Attributes;
+
+[AttributeUsage(AttributeTargets.Class)]
+public class SchoolAuthorizationAttribute : AuthorizeAttribute, IAuthorizationFilter
+{
+    public void OnAuthorization(AuthorizationFilterContext context)
+    {
+        var urn = context.RouteData.Values["urn"]?.ToString();
+
+        var isValid = context.HttpContext.User.Claims.Any(c => urn != null && c.Type == "Schools" && c.Value.Contains(urn));
```

Comment:
Do we need null checks on `User.Claims` always or `Claim.Value` or is this safe enough as-is?

Reply:
Should be enough as-is. But we can ensure we have integration tests to cover this when we fully integrate DSI.